src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.cs

```diff
@@ -0,0 +1,109 @@
+
+        public void Execute(GeneratorExecutionContext context)
+        {
+            var receiver = context.SyntaxReceiver as SyntaxReceiver;
+            if (receiver is null || receiver.CandidateClasses is null || receiver.CandidateClasses.Count == 0)
+            {
+                // nothing to do yet
+                return;
+            }
+
+            var p = new Parser(context.Compilation, context.ReportDiagnostic, context.CancellationToken);
+            var eventSources = p.GetEventSourceClasses(receiver.CandidateClasses);
+
+            if (eventSources?.Length > 0)
+            {
+                var e = new Emitter(context);
+                e.Emit(eventSources, context.CancellationToken);
+            }
+        }
+
+        private sealed class SyntaxReceiver : ISyntaxReceiver
+        {
+            private List<ClassDeclarationSyntax>? _candidateClasses;
+
+            public List<ClassDeclarationSyntax>? CandidateClasses => _candidateClasses;
+
+            public void OnVisitSyntaxNode(SyntaxNode syntaxNode)
+            {
+                // Only add classes annotated [EventSourceAutoGenerate] to reduce busy work.
+                const string EventSourceAttribute = "EventSourceAutoGenerateAttribute";
+                const string EventSourceAttributeShort = "EventSourceAutoGenerate";
+
+                // Only clasess
+                if (syntaxNode is ClassDeclarationSyntax classDeclaration)
+                {
+                    // Check if has EventSource attribute before adding to candidates
+                    // as we don't want to add every class in the project
+                    foreach (var cal in classDeclaration.AttributeLists)
+                    {
+                        foreach (var ca in cal.Attributes)
+                        {
+                            // Check if Span length matches before allocating the string to check more
+                            int length = ca.Name.Span.Length;
+                            if (length != EventSourceAttribute.Length && length != EventSourceAttributeShort.Length)
+                            {
+                                continue;
+                            }
+
+                            // Possible match, now check the string value
+                            string attrName = ca.Name.ToString();
+                            if (attrName == EventSourceAttribute || attrName == EventSourceAttributeShort)
+                            {
+                                // Match add to candidates
+                                _candidateClasses ??= new List<ClassDeclarationSyntax>();
+                                _candidateClasses.Add(classDeclaration);
+                                return;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // Can change to terse record syntax as isn't supported by netstandard 2.0
+        private record EventSourceClass
```

Comment:
Records are bloatware. They come with a pile of auto-generated stuff that we do not need. I think we should stick with classic classes in the core framework.

Reply:
Reverted